cowamm/kpi/competitors/curve/curve_kpis_4232873.sql

```diff
@@ -29,8 +24,28 @@ from (
         on
             r.contract_address = t.project_contract_address
             and r.tx_hash = t.tx_hash
+    inner join prices.minute as p0
+        on
+            r.token0 = p0.contract_address
+            and date_trunc('minute', r.evt_block_time) = p0.timestamp
+    inner join prices.minute as p1
+        on
+            r.token1 = p1.contract_address
+            and date_trunc('minute', r.evt_block_time) = p1.timestamp
     where
         -- This test avoids any possible issue with reconstructing the reserves of the pool
         tvl > 0
+        and p0.timestamp >= date_add('day', -1, (case when '{{competitor_end_time}}' = '2100-01-01' then now() else timestamp '{{competitor_end_time}}' end))
+        and p0.timestamp <= (case when '{{competitor_end_time}}' = '2100-01-01' then now() else timestamp '{{competitor_end_time}}' end)
```

Comment:
Maybe using 
```
p0.timestamp between date_add('day', -1, least('{{competitor_end_time}}', now())) and least('{{competitor_end_time}}', now())
```

is slightly easier to read? Also I'm not sure if reducing the range to `now()` is actually making the query more effective (since there are no datapoints beyond now the full table scan should complete in the same time)